spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -1052,9 +1039,10 @@ public synchronized void testDeleteWithSerializableIsolation() throws Interrupte
         executorService.submit(
             () -> {
               for (int numOperations = 0; numOperations < Integer.MAX_VALUE; numOperations++) {
-                while (barrier.get() < numOperations * 2) {
-                  sleep(10);
-                }
+                int currentNumOperations = numOperations;
+                Awaitility.await()
+                    .pollInterval(Duration.ofMillis(10))
```

Comment:
I think all of these should have `.atMost(5, TimeUnit.SECONDS)` because we don't want to wait indefinitely for the condition to happen

Reply:
Thanks for the suggestion. I think the default value of atMost with 10 seconds is applied to all the Awaitility parts. But, as your comment, I believe it's better to explicitly specify the timeout to all of them. I will do it (if it's no need, please let me know).

Reply:
Updated.